samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -16,173 +16,109 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package org.apache.samza.operators;
 
 import org.apache.samza.annotation.InterfaceStability;
 import org.apache.samza.operators.data.Message;
-import org.apache.samza.operators.internal.Operators;
-import org.apache.samza.operators.internal.Operators.Operator;
-import org.apache.samza.operators.internal.WindowOutput;
-import org.apache.samza.task.MessageCollector;
-import org.apache.samza.task.TaskCoordinator;
+import org.apache.samza.operators.functions.FilterFunction;
+import org.apache.samza.operators.functions.FlatMapFunction;
+import org.apache.samza.operators.functions.JoinFunction;
+import org.apache.samza.operators.functions.MapFunction;
+import org.apache.samza.operators.functions.SinkFunction;
+import org.apache.samza.operators.windows.Window;
+import org.apache.samza.operators.windows.WindowOutput;
+import org.apache.samza.operators.windows.WindowState;
 
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.Set;
-import java.util.function.BiFunction;
-import java.util.function.Function;
 
 
 /**
- * This class defines either the input or output streams to/from the operators. Users use the API methods defined here to
- * directly program the stream processing stages that processes a stream and generate another one.
- *
- * @param <M>  Type of message in this stream
+ * Represents a stream of values of type {@link Message}. A {@link MessageStream} can be transformed into another
+ * {@link MessageStream} by applying the transforms in this API.
+ * @param <M>  type of {@link Message}s in this stream
  */
 @InterfaceStability.Unstable
-public class MessageStream<M extends Message> {
-
-  private final Set<Operator> subscribers = new HashSet<>();
-
-  /**
-   * Helper method to get the corresponding list of subscribers to a specific {@link MessageStream}.
-   *
-   * NOTE: This is purely an internal API and should not be used directly by programmers.
-   *
-   * @return A unmodifiable set containing all {@link Operator}s that subscribe to this {@link MessageStream} object
-   */
-  public Collection<Operator> getSubscribers() {
-    return Collections.unmodifiableSet(this.subscribers);
-  }
-
-  /**
-   * Public API methods start here
-   */
+public interface MessageStream<M extends Message> {
 
   /**
-   * Defines a function API that takes three input parameters w/ types {@code A}, {@code B}, and {@code C} and w/o a return value
+   * Applies the 1:1 {@link MapFunction} to this {@link MessageStream} and returns the resulting {@link MessageStream}.
```

Comment:
Yeah, will make them consistent. Slightly prefer the more explicit/descriptive "{@link Messages} in this {@link MessageStream}". Thoughts?